File: src/main/java/com/skidders/sigma/module/settings/impl/ModeSetting.java
package com.skidders.sigma.module.settings.impl;

import com.skidders.sigma.module.settings.Setting;

public class ModeSetting extends Setting<String> {

    public String[] allValues;

    public ModeSetting(String name, String desc, String value, String[] allValues) {
        super(name, desc, value);
        this.allValues = allValues;
    }
}

File: src/main/kotlin/com/skidders/sigma/module/settings/impl/ModeSetting.kt
package com.skidders.sigma.module.settings.impl

import com.skidders.sigma.module.settings.Setting

class ModeSetting : Setting<String> {

    val allValues: Array<String>

    constructor(name: String, desc: String, value: String, allValues: Array<String>) : super(name, desc, value) {
        this.allValues = allValues
    }
}
